### PR TITLE
Add eslint

### PR DESCRIPTION
**Pull request by Hinton**
## Type of change

- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

Initial work at replacing tslint with eslint. I went with the default eslint rules and added some of the old rules as well. I didn't add `member-ordering`, I'm open to adding it, but we disable it in a few places already since js needs to be organized a specific way, so unsure how much value it would provide.

I mainly preferred to ignore eslint errors vs fixing them. We should remove any comments when we modify nearby code but it seemed a waste of time to test everything now.

Biggest change to before is that imports are now automatically sorted and grouped using eslint. This results in a slightly changed grouping than we are used to. I think the benefits of automating it is an improvement.

## Before you submit

- [ ] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

**Comment by Hinton** [comment]
Added hold to make sure we don't merge this before `rc` is cut.

File: common/src/models/request/tokenRequest.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import { TwoFactorProviderType } from "../../enums/twoFactorProviderType";
+import { Utils } from "../../misc/utils";
+
+import { CaptchaProtectedRequest } from "./captchaProtectedRequest";
+import { DeviceRequest } from "./deviceRequest";
+
+export class TokenRequest implements CaptchaProtectedRequest {
```
Comment (jlf0dev) [inline common/src/models/request/tokenRequest.ts:7]: Curious about this added file?

Comment (Hinton) [reply common/src/models/request/tokenRequest.ts:7]: Probably a file that was removed in a PR and got accidently re-added.

**Review by jlf0dev — APPROVED** [review]

**Comment by jlf0dev** [comment]
I found one missing reference to `tslint` in `bitwardenpasswordprotectedimporter.ts` in your latest merge from master.

Other than that looks good!

**Review by jlf0dev — APPROVED** [review]

**Pull request merged**